### PR TITLE
Make the `sel` error more descriptive when `method` is unset

### PR DESCRIPTION
**Pull request by jsignell**
- [x] Tests added

This came up in the tutorial and I was wondering if the error could be made a little clearer. Not sure if the error message should hint that a user might want to use ``method``.

```python
import numpy as np
import pandas as pd
import xarray as xr

arr = xr.DataArray(
    data=np.arange(48).reshape(4, 2, 6),
    dims=("u", "v", "time"),
    coords={
        "u": [-3.2, 2.1, 5.3, 6.5],
        "v": [-1, 2.6],
        "time": pd.date_range("2009-01-05", periods=6, freq="M"),
    },
)
arr.sel(u=5, time="2009-04-28")  # I removed `method="nearest"`
```

Before this PR:

```python-traceback
...
KeyError: 5.0
```

After this PR:

```python-traceback
...
KeyError: "not all values found in index 'u'. Did you mean to set `method=`?"
```

### CODE REVIEW

File: xarray/core/indexes.py [hunk 1]
```diff
@@ -404,7 +404,14 @@ def sel(
                                 f"not all values found in index {coord_name!r}"
                             )
                     else:
-                        indexer = self.index.get_loc(label_value)
+                        try:
+                            indexer = self.index.get_loc(label_value)
+                        except KeyError as e:
+                            raise KeyError(
+                                f"not all values found in index {coord_name!r}. "
+                                "Did you mean to set `method=`?"
```
Comment (dcherian) [inline xarray/core/indexes.py:412]:
```suggestion
                                f"not all values found in index {coord_name!r}. "
                                "Try setting the `method` keyword argument (example: method='nearest')."
```

File: xarray/tests/test_dataarray.py [hunk 2]
```diff
@@ -1055,6 +1055,9 @@ def test_sel_no_index(self) -> None:
     def test_sel_method(self) -> None:
         data = DataArray(np.random.randn(3, 4), [("x", [0, 1, 2]), ("y", list("abcd"))])
 
+        with pytest.raises(KeyError, match="Did you mean to set `method=`?"):
```
Comment (dcherian) [inline xarray/tests/test_dataarray.py:1058]:
```suggestion
        with pytest.raises(KeyError, match="Try setting the `method`"):
```

**Review by dcherian — APPROVED** [review]
Thanks @jsignell This is a nice improvement!

**Comment by mathause** [comment]
Thanks @jsignell!

**Pull request merged**